python/tank/platform/bundle.py

```diff
@@ -221,15 +231,24 @@ def cache_location(self):
         
         return self.__cache_location
 
-    @property
-    def context(self):
+    def _get_context(self):
         """
-        The current context associated with this item
+        The current context associated with this item.
         
         :returns: context object
         """
         return self.__context
 
+    def _set_context(self, new_context):
+        """
+        Sets the current context associated with this item.
```

Comment:
I think this is potentially confusing. If I understand correctly, accessing `self.context` to retrieve the context is common in apps. But *setting* it should only happen during the context switch right? So this is always called via an internal method, in a controlled fashion, during the context switch. I think maybe this shouldn't be a setter on a property but an internal method instead for clarity. Does that make sense? I am right thinking that the only thing that can update the context is the context switching logic?

Similarly, I think the setter for `settings` above may be similar? Can settings be updated anytime or does it need to happen as part of the switch? If the latter, then it may be better to have a separate method for it?